### ISSUE TITLE
Chime not implemented in RFXCOM binding

### ISSUE THREAD

**Issue by sheppy99** [openhab/openhab#3058]
Can the Chime function be implemented please in the RFXCOM binding so a doorbell can be used as an input into OpenHAB?
It's enabled in bit 5 in the msg3 byte - represented by "2" in the initialization string "0D00000003530020000500000000"
The error message in debug mode is:
10:53:53.840 [DEBUG] [.b.r.internal.RFXComConnection:151  ] - Unknown data received, data: 0716000000160170
10:53:53.872 [ERROR] [.rfxcom.internal.RFXComBinding:292  ] - Error occured during packet receiving, data: 0716000000160170
org.openhab.binding.rfxcom.internal.RFXComException: Message CHIME not implemented

**Comment by Speederc** [comment]
Hi Sheppy99,
I prepared 2 files to add this CHIME support but unfortunately I'm not able to compile them before next week. Are you able to do it ? If yes the files are there :
https://github.com/Speederc/openhab/blob/d1a33211fbddffc703d25ae0e04b8f233d7ff5b8/bundles/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComChimeMessage.java
https://github.com/Speederc/openhab/blob/d1a33211fbddffc703d25ae0e04b8f233d7ff5b8/bundles/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/RFXComValueSelector.java
I hope that it works ! The value can be retrieve by the item using "ChimeSound"

**Comment by sheppy99** [comment]
Thanks very much for working on this, I can have a go at a compile if there is a wiki showing me what to do or I'm happy to wait until next week, will leave it up to you.

**Comment by Speederc** [comment]
The openhab compilation is described there : https://github.com/openhab/openhab/wiki/IDE-Setup . I prefer that you compile and test on your system to be sure that it works.

**Comment by sheppy99** [comment]
OK, I'll try and get onto it over the weekend - I live down under so I'm off to bed shortly.

**Comment by sheppy99** [comment]
I've just skim read the compilation procedure, does it have to be done on Linux? The only Linux machine I have is a Pi2 which could take some time!

**Comment by sheppy99** [comment]
Sorry to say I've drawn a blank on this one. I tried to use the instructions to create an IDE using the Vagrant method, that didn't work as now Vagrant has become Atlas and the machine is no longer available. SO with a spot of googling I managed to get a Windows Emulation of a Raspberry Pi as I'm concerned about adding extra stuff to my Pi which I might not be able to get rid of. Whilst this booted it sadly doesn't under stand the command "GIT." This is probably an easy fix for someone who knows Linux which I don't and  sadly I don't have the time left this weekend to spend on it.

I'll happily extensively test it for you if only someone can either give me an easy way to compile this under Win7/64 or compile it for me.

Last time I had the rfxcom plugged into windows I noticed the odd chime packet around here so I'm not the only one with a doorbell!

**Comment by Speederc** [comment]
Please test with this version : http://marionpetelot.fr/temp/org.openhab.binding.rfxcom_1.8.0.201508312140.jar . It adds UV, attributes of TFA_WIND and Chime messages. If it works, I will add it in official binding.

**Comment by sheppy99** [comment]
Binding loaded and seems to be working for existing items. For Chime I added the following to my items
String Doorbell { rfxcom="<22.1:RawData" }
and to my sitemap
Text item=Doorbell label="Doorbell [%s]"
I can see in the log
Raw data = 0716000F00160160
 - Packet type = CHIME
 - Seq number = 15
 - Sub type = BYRONSX
 - Id = 22
 - Chime Sound = 1
 - Signal level = 6
 - Battery level = 0
but nothing changes on my sitemap when I press the doorbell.
I was expecting to see the rawdata so I could work out which parameter to use. Should I use something different in my items?

**Comment by Speederc** [comment]
I forgot to mention that you can use "ChimeSound" type as a number in your item declaration

**Comment by sheppy99** [comment]
I thought that was the way to use it
{ rfxcom="<Id.Chime Sound:RawData" }
but I don't have anything that appears in debug log apart from the received message

**Comment by Speederc** [comment]
There is no sensorId for Chime sensor, item should look like : Number DoorSound { rfxcom="<22:ChimeSound" }

**Comment by sheppy99** [comment]
String Doorbell { rfxcom="<22:ChimeSound" }
gives the following error in the log
2015-09-01 19:26:32.190 [ERROR] [i.internal.GenericItemProvider] - Binding configuration of type 'rfxcom' of item ?Doorbell? could not be parsed correctly.
org.openhab.model.item.binding.BindingConfigParseException: Invalid item type for value selector 'ChimeSound'!

**Comment by sheppy99** [comment]
Number Doorbell { rfxcom="<22:ChimeSound" } in default.items
Text item=Doorbell label="Doorbell [%.0f]" in default.sitemap

Gives "1" all the time after the first bell push - do I need a rule to reset it?

**Comment by Speederc** [comment]
I don't know how the Chime sensor works. It seems that it only sends a message for switching on, not for switching off. Thus you probably have to create a rule with " Item <item> received command [<command>]" to detect the reception of a new Chime.

**Comment by sheppy99** [comment]
OK, I'll have a play with rules tomorrow to see how I can get it to work in practice and let you know.
One other thing I've noticed with the system is it fails to transmit when it is receiving packets. I have Home Easy PIR's and they transmit 2 codes when they trigger, Lighting2 and then Lighting1 formats, and I have to use the last command with a rule as using the first one causes the rule to fail.
If this a limit of the rfxcom device?
If I turn off receiving one type can I still transmit it?

**Comment by sheppy99** [comment]
I've got a rule that makes it work but I can't use the binding to send a message to the doorbell - can you advise the syntax for that? I've tried many but none seem to work.

For receiving - relies on postUpdate(Doorbell, 0) on startup:
rule "Doorbell"
when Item Doorbell changed
then
	if (Doorbell.state == 1)
{	Doorbellpressed = createTimer(now.plusSeconds(5))
[|	postUpdate(Doorbell, 0)	]}
end

**Comment by Speederc** [comment]
Sorry not idea for that

**Comment by sheppy99** [comment]
Can it be added to the binding if I provide the packet details from the Windows rfxmngr app?

**Comment by Speederc** [comment]
What do you want to add ?

**Comment by sheppy99** [comment]
I would like to send chime messages as well as receive them.  They seem to take the form of
ID which is a number and CHIME NOTE which also is a number. The type is ByronSX

**Comment by sheppy99** [comment]
A further update on this one. I've had sat on my desk a bell push and a bell and it seems no matter where I change the code to that there are always false alarms. In my case this means I can't use the product. I am happy to test any further work but without a way of changing to another code that isn't in use around here the chime for me isn't going to work.

**Comment by AlexCPU** [comment]
Hi @Speederc,
I've had a bit of a play with this, and the receiving of CHIME messages seems to work successfully (I'm testing with a ByronSX compatible system).
My items file contains:
Number DoorbellButton {rfxcom="<53:ChimeSound"}
I have set MySQL persistence up to log every update from this sensor and therefore I can see a log entry every time the doorbell is pushed.  The ChimeSound represents the ID of the sound that the doorbell receiver should play, so generally it is always the same.

What I (and I think @sheppy99) can't work out how to do is send that message such that I can make the doorbell sound (and therefore use the different tones to mean different things - some of the ByronSX's allow you to load MP3s for each sound).  Using RFXmngr I can select the Chime tab, select ByronSX, select the ID and the sound, and when I send it, the chime is played, which as a binding I would expect to look something like this:
Number DoorbellSounder {rfxcom=">53.0:Chime.ByronSX:ChimeSound"}

Any suggestions what this binding should look like?

**Comment by Speederc** [comment]
Hi @AlexCPU,
I tried to generate a new version with this "Chime output" support : http://dr-petelot-marion.chirurgiens-dentistes.fr/temp/org.openhab.binding.rfxcom_1.9.0.201601221259.jar .
Your item should look like : Number DoorbellSounder {rfxcom=">53:ChimeSound"} and you should provide it the chime sound index (for ByronSX it seems to be 1,2,3 or 5).
Can you test it and give me your feedback ?
Thanks

**Comment by McShannon** [comment]
Hi @Speederc
Hope you don't mind, i downloaded the new binding and the input works great. I cannot get the output to work. I am unsure of the format. {rfxcom=">53:ChimeSound"} is what i have used but the output is looking for a second part like @AlexCPU showed before {rfxcom=">53.0:Chime.ByronSX:ChimeSound"}
Thanks for all your efforts thus far.

**Comment by Speederc** [comment]
Hi @McShannon 
I agree that "Chime.ByronSX:" has to be added in the item description but not the ".0" which is useless in that case. Can you try it ? Can you also send me the openhab log when you send the command ?

**Comment by McShannon** [comment]
Hi @Speederc 
Here is the item config.....

Switch 	DoorbellSounder {rfxcom=">53:Chime.ByronSX:ChimeSound"}

Here is the output when it reloads the items file....

09:43:51.332 [ERROR] [i.internal.GenericItemProvider:350  ] - Binding configuration of type 'rfxcom' of item ‘DoorbellSounder‘ could not be parsed correctly.
org.openhab.model.item.binding.BindingConfigParseException: Invalid packet type 'Chime'!
	at org.openhab.binding.rfxcom.internal.RFXComGenericBindingProvider.processBindingConfiguration(RFXComGenericBindingProvider.java:110) ~[na:na]
	at org.openhab.model.item.internal.GenericItemProvider.internalDispatchBindings(GenericItemProvider.java:348) [org.openhab.model.item_1.7.1.jar:na]
	at org.openhab.model.item.internal.GenericItemProvider.internalDispatchBindings(GenericItemProvider.java:324) [org.openhab.model.item_1.7.1.jar:na]
	at org.openhab.model.item.internal.GenericItemProvider.processBindingConfigsFromModel(GenericItemProvider.java:171) [org.openhab.model.item_1.7.1.jar:na]
	at org.openhab.model.item.internal.GenericItemProvider.modelChanged(GenericItemProvider.java:390) [org.openhab.model.item_1.7.1.jar:na]
	at org.openhab.model.core.internal.ModelRepositoryImpl.notifyListeners(ModelRepositoryImpl.java:159) [org.openhab.model.core_1.7.1.jar:na]
	at org.openhab.model.core.internal.ModelRepositoryImpl.addOrRefreshModel(ModelRepositoryImpl.java:100) [org.openhab.model.core_1.7.1.jar:na]
	at org.openhab.model.core.internal.folder.FolderObserver.checkFolder(FolderObserver.java:142) [org.openhab.model.core_1.7.1.jar:na]
	at org.openhab.model.core.internal.folder.FolderObserver.run(FolderObserver.java:99) [org.openhab.model.core_1.7.1.jar:na]

**Comment by Speederc** [comment]
It seems that the type and subtype have to be written in upper case. Please try with  {rfxcom=">53:CHIME.BYRONSX:ChimeSound"} . Can you also send me the openhab log when you send the command ?

**Comment by McShannon** [comment]
Hi @Speederc, hope this helps...

{rfxcom=">53:CHIME.BYRONSX:ChimeSound"} 

returns

09:45:20.852 [ERROR] [i.internal.GenericItemProvider:350  ] - Binding configuration of type 'rfxcom' of item â€˜Bellâ€˜ could not be parsed correctly.
org.openhab.model.item.binding.BindingConfigParseException: Invalid item type for value selector 'ChimeSound'!
	at org.openhab.binding.rfxcom.internal.RFXComGenericBindingProvider.processBindingConfiguration(RFXComGenericBindingProvider.java:138) ~[na:na]
	at org.openhab.model.item.internal.GenericItemProvider.internalDispatchBindings(GenericItemProvider.java:348) [org.openhab.model.item_1.8.0.201509080234.jar:na]
	at org.openhab.model.item.internal.GenericItemProvider.internalDispatchBindings(GenericItemProvider.java:324) [org.openhab.model.item_1.8.0.201509080234.jar:na]
	at org.openhab.model.item.internal.GenericItemProvider.processBindingConfigsFromModel(GenericItemProvider.java:171) [org.openhab.model.item_1.8.0.201509080234.jar:na]
	at org.openhab.model.item.internal.GenericItemProvider.modelChanged(GenericItemProvider.java:390) [org.openhab.model.item_1.8.0.201509080234.jar:na]
	at org.openhab.model.core.internal.ModelRepositoryImpl.notifyListeners(ModelRepositoryImpl.java:159) [org.openhab.model.core_1.8.0.201509080234.jar:na]
	at org.openhab.model.core.internal.ModelRepositoryImpl.addOrRefreshModel(ModelRepositoryImpl.java:100) [org.openhab.model.core_1.8.0.201509080234.jar:na]
	at org.openhab.model.core.internal.folder.FolderObserver.checkFolder(FolderObserver.java:142) [org.openhab.model.core_1.8.0.201509080234.jar:na]
	at org.openhab.model.core.internal.folder.FolderObserver.run(FolderObserver.java:99) [org.openhab.model.core_1.8.0.201509080234.jar:na]

**Comment by Speederc** [comment]
Hi @McShannon,
You have to use a Number type item for this command as sound chime index (different sounds I guess : 1,2,3 or 5 for BYRONSX) has to be provided : Number DoorbellSounder {rfxcom=">53:CHIME.BYRONSX:ChimeSound"} . If you want to use a switch item to control your doorbell, you will have to do it using a rule which links your Switch item to the Number item.

**Comment by McShannon** [comment]
Hi @Speederc
Let me try and understand whats happening. I have a number item setup that will receive all incoming button presses. When the physical doorbell button is pressed it tells me what chime number it received. There is a button on the back of the physical doorbell button that allows me to change what chime is played. Changing the chime on the back then pushing the button on the front changes the number in the following:

Number DoorbellButton {rfxcom="<53:ChimeSound"} This currently shows 14 (Options are 1,2,3,5,6,9,13,14). Openhab receives the incoming message "<" and displays the result "14"

If i want to send a command to the doorbell using a virtual doorbell button I have to send the command along with a number so that the doorbell knows what chime to play. ">" and play "3"

As the doorbell is capable of playing multiple tunes I wanted it to play a specific tune when the external PIR is triggered (3) and a different tune when someone pushes the physical doorbell button (14).  

I assumed that I would have to use a switch to emulate the physical doorbell button and when the switch is triggered a command is sent out via {rfxcom=">53:CHIME.BYRONSX:ChimeSound"} I do not know how to send out a number?

I apologise if I am causing you confusion and i do really appreciate your assistance.

**Comment by Speederc** [comment]
Hi @McShannon,

Your understanding is correct. In your case there will be 2 different items :
```
Number DoorbellButton {rfxcom="<53:ChimeSound"} //input : get the chime sound index when the external button is pressed (this will be used only as info, to know what is the sound index for a specific chime)
Number DoorbellSounder {rfxcom=">53:CHIME.BYRONSX:ChimeSound"} //output : can generate a RFXCOM message embedding the chime sound index
```

In your rule file, there will be something like : 
```
rule "Chime linked to PIR"
when
    Item PIRitem changed from OFF to ON
then
    sendCommand(DoorbellSounder , 3) // sending 3 to this item will generated a RFXCOM message with this chime index
end
```
I can not test it on my side, as I don't have this kind of doorbell, but this is how I would do it !

**Comment by McShannon** [comment]
Hi @Speederc 

Now that I actually understand what is happening here are my results...

Here is my rule(I used an internal door as it was easier to test)

rule "Cave Door"
when
	Item FF_Front_Door changed to ON
then
	sendCommand(DoorbellSounder, 14)
end

When i trigger it I get this...

2016-01-25 20:04:07.477 [INFO ] [runtime.busevents             ] - FF_Front_Door received command ON
2016-01-25 20:04:07.483 [WARN ] [.rfxcom.internal.RFXComBinding] - Provider is not in binding 'org.openhab.binding.rfxcom.internal.RFXComGenericBindingProvider@6a744c2'
2016-01-25 20:04:07.810 [INFO ] [runtime.busevents             ] - DoorbellSounder received command 14

Doorbell does not ring.

Many Thanks

**Comment by Speederc** [comment]
@McShannon, this log seems to indicate that your DoorbellSounder is a input item. Can you check that you use ">" for this item ? If it is the case, can you send me your item, rules and log file ? It will be easier to investigate.

**Comment by McShannon** [comment]
Hi @Speederc, thanks for your patience.

Here are the items

`Number DoorbellButton "Chime Received [%.1f]" {rfxcom="<53:ChimeSound"}`
`Number DoorbellSounder "Chime Sent [%.1f]" {rfxcom=">53:CHIME.BYRONSX:ChimeSound"}`

Here is the rule

`rule "Cave Door"`
`when`
	`Item FF_Front_Door changed to ON`
`then`
	`sendCommand(DoorbellSounder,14)`
`end`

Here is the log

`17:28:12.806 [DEBUG] [o.i.r.i.resources.ItemResource:193  ] - Received HTTP POST request at 'items/FF_Front_Door' with value 'ON'.`
`17:28:12.807 [DEBUG] [.rfxcom.internal.RFXComBinding:94   ] - Received command: FF_Front_Door ON`
`17:28:12.808 [WARN ] [.rfxcom.internal.RFXComBinding:168  ] - Provider is not in binding 'org.openhab.binding.rfxcom.internal.RFXComGenericBindingProvider@55d8ac0f'`
`17:28:12.809 [DEBUG] [m.r.internal.engine.RuleEngine:305  ] - Executing rule 'Cave Door'`
`17:28:12.815 [DEBUG] [.rfxcom.internal.RFXComBinding:94   ] - Received command: DoorbellSounder 14`
`17:28:12.815 [DEBUG] [.rfxcom.internal.RFXComBinding:152  ] - Received command (item='DoorbellSounder', state='14', class='class org.openhab.core.library.types.DecimalType')`
`17:28:12.824 [DEBUG] [.m.i.MongoDBPersistenceService:161  ] - MongoDB save FF_Front_Door=ON`
`17:28:12.825 [INFO ] [runtime.busevents             :22   ] - FF_Front_Door received command ON`
`17:28:12.842 [INFO ] [runtime.busevents             :22   ] - DoorbellSounder received command 14`
`17:28:13.123 [DEBUG] [o.o.i.r.i.b.GeneralBroadcaster:46   ] - broadcaster 'org.openhab.io.rest.internal.broadcaster.GeneralBroadcaster$1@7629d1ff' is empty`

**Comment by Speederc** [comment]
Hi @McShannon 
It is hard to say with this log. I created a new version which adds more log details : http://dr-petelot-marion.chirurgiens-dentistes.fr/temp/org.openhab.binding.rfxcom_1.9.0.201601262305.jar . Can you test it ? Can you also send me your full (not an extract) item and logs files through a storage website (Gdrive, dropbox,...) ?

**Comment by McShannon** [comment]
Hi @Speederc 

I have created a test environment here at work using a raspberry pi and a spare RFXCOM box. I will set it up when i get home and send you the results. I reckon a clean build with no other factors will make fault finding easier.

Thanks again.

**Comment by McShannon** [comment]
Hi @Speederc 

I have uploaded the files. Let me know if you need more info.

https://www.dropbox.com/s/3dn6txoym0ywqi6/RFXCOM.rar?dl=0

Many Thanks

Tom

**Comment by AlexCPU** [comment]
I can confirm that mine is behaving exactly the same as @McShannon is seeing.  
Output in the log shows as below, which looks the same, all my items, rules, sitemap match his too:
```
2016-01-27 21:19:40.670 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command: DoorbellSounder 3
2016-01-27 21:19:40.673 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command (item='DoorbellSounder', state='3', class='class org.openhab.core.library.types.DecimalType')
2016-01-27 21:19:40.674 [DEBUG] [.rfxcom.internal.RFXComBinding] - RFXComSerialConnector
2016-01-27 21:19:40.676 [DEBUG] [.rfxcom.internal.RFXComBinding] - Pre executeCommand0 (item=DoorbellSounder, command=3)
2016-01-27 21:19:40.677 [DEBUG] [.rfxcom.internal.RFXComBinding] - executeCommand0 (item=DoorbellSounder, command=3)
2016-01-27 21:19:40.679 [DEBUG] [.rfxcom.internal.RFXComBinding] - executeCommand0 valueSelector (item=DoorbellSounder, command=3)
```

Just as a matter of curiosity Tom, does your doorbell actually have the same ID of 53 as mine does?  I've got two systems, a really old one, which has an ID of 60 (I replaced it as the push button is dying), and a new one in the LightwaveRF range with an ID of 53.  Neither of them work using openHAB, and I've only tested the old one (ID 60) using rfxmngr, but that worked fine at the time.

**Comment by Speederc** [comment]
Hi @McShannon and @AlexCPU
Still hard to say where the message is blocked. Everything seems to go smoothly up to few lines before the transmission. I created a new version with again more log details :  http://dr-petelot-marion.chirurgiens-dentistes.fr/temp/org.openhab.binding.rfxcom_1.9.0.201601272254.jar . Can you test it ?
Sorry for this iterative debugging...

**Comment by AlexCPU** [comment]
That seems to be quite a bit broken, as now the lights don't work either!
```
2016-01-27 22:05:29.488 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command: LoungeLight ON
2016-01-27 22:05:29.493 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command (item='LoungeLight', state='ON', class='class org.openhab.core.library.types.OnOffType')
2016-01-27 22:05:29.494 [DEBUG] [.rfxcom.internal.RFXComBinding] - RFXComSerialConnector
2016-01-27 22:05:29.495 [DEBUG] [.rfxcom.internal.RFXComBinding] - executeCommand0 (item=LoungeLight, command=ON)
2016-01-27 22:05:29.496 [DEBUG] [.rfxcom.internal.RFXComBinding] - 1-executeCommand0 (item=LoungeLight, id=2.1, packetType=LIGHTING5, subType=LIGHTWAVERF, valueSelector=DimmingLevel)
2016-01-27 22:05:29.497 [DEBUG] [.rfxcom.internal.RFXComBinding] - 2-executeCommand0 (item=LoungeLight, command=ON)
```

And the log from the doorbell is:
```
2016-01-27 22:07:48.280 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command: DoorbellSounder 3
2016-01-27 22:07:48.281 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command (item='DoorbellSounder', state='3', class='class org.openhab.core.library.types.DecimalType')
2016-01-27 22:07:48.282 [DEBUG] [.rfxcom.internal.RFXComBinding] - RFXComSerialConnector
2016-01-27 22:07:48.283 [DEBUG] [.rfxcom.internal.RFXComBinding] - executeCommand0 (item=DoorbellSounder, command=3)
2016-01-27 22:07:48.284 [DEBUG] [.rfxcom.internal.RFXComBinding] - 1-executeCommand0 (item=DoorbellSounder, id=53, packetType=CHIME, subType=BYRONSX, valueSelector=ChimeSound)
2016-01-27 22:07:48.285 [DEBUG] [.rfxcom.internal.RFXComBinding] - 2-executeCommand0 (item=DoorbellSounder, command=3)
```

**Comment by Speederc** [comment]
Again a new version with logs :  http://dr-petelot-marion.chirurgiens-dentistes.fr/temp/org.openhab.binding.rfxcom_1.9.0.201601280033.jar

**Comment by McShannon** [comment]
Hi @AlexCPU yes, my bell has an ID of 53. It works well in RFXManager.
Hi @Speederc, I will give the new one a go this evening.

**Comment by AlexCPU** [comment]
Here you go, latest version log:
```
2016-01-28 18:18:29.965 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command: DoorbellSounder 3
2016-01-28 18:18:29.969 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command (item='DoorbellSounder', state='3', class='class org.openhab.core.library.types.DecimalType')
2016-01-28 18:18:29.972 [DEBUG] [.rfxcom.internal.RFXComBinding] - RFXComSerialConnector
2016-01-28 18:18:29.976 [DEBUG] [.rfxcom.internal.RFXComBinding] - executeCommand0 (item=DoorbellSounder, command=3)
2016-01-28 18:18:29.978 [DEBUG] [.rfxcom.internal.RFXComBinding] - 1-executeCommand0 (item=DoorbellSounder, id=53, packetType=CHIME, subType=BYRONSX, valueSelector=ChimeSound)
2016-01-28 18:18:29.979 [DEBUG] [.rfxcom.internal.RFXComBinding] - 2-executeCommand0 (item=DoorbellSounder, command=3)
```
Lighting is still broken too
```
2016-01-28 18:19:36.647 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command: LoungeLight OFF
2016-01-28 18:19:36.650 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command (item='LoungeLight', state='OFF', class='class org.openhab.core.library.types.OnOffType')
2016-01-28 18:19:36.651 [DEBUG] [.rfxcom.internal.RFXComBinding] - RFXComSerialConnector
2016-01-28 18:19:36.652 [DEBUG] [.rfxcom.internal.RFXComBinding] - executeCommand0 (item=LoungeLight, command=OFF)
2016-01-28 18:19:36.653 [DEBUG] [.rfxcom.internal.RFXComBinding] - 1-executeCommand0 (item=LoungeLight, id=2.1, packetType=LIGHTING5, subType=LIGHTWAVERF, valueSelector=DimmingLevel)
2016-01-28 18:19:36.653 [DEBUG] [.rfxcom.internal.RFXComBinding] - 2-executeCommand0 (item=LoungeLight, command=OFF)
```

**Comment by Speederc** [comment]
Thank you for your tests. Nothing explains this behavior. I am finally uncertain about the branch 1.9.0 sources files or my compilation environment. Can you test this version which is a pure 1.9.0 branch (without the Chime modifications) : http://dr-petelot-marion.chirurgiens-dentistes.fr/temp/org.openhab.binding.rfxcom_1.9.0.branch.jar . Can you confirm that lighting2 (not chime) works with this one ?

**Comment by McShannon** [comment]
Hi @Speederc 

Lighting2 works fine with this one.

**Comment by Speederc** [comment]
Hi @McShannon , @AlexCPU 
I generated a new clean version with only Chime added : http://dr-petelot-marion.chirurgiens-dentistes.fr/temp/org.openhab.binding.rfxcom_1.9.0.201601292322.jar . Could you test it ?

**Comment by McShannon** [comment]
Hi @Speederc 

Here is what we are getting now that is different from before. I have changed the original item name from doorbellsounder to bell which is why the log shows item=bell. Thats all the log shows and the bell does not chime.

14:39:17.048 [DEBUG] [m.r.internal.engine.RuleEngine:305  ] - Executing rule 'Cave Door'
14:39:17.564 [DEBUG] [.rfxcom.internal.RFXComBinding:94   ] - Received command: Bell 14
14:39:17.577 [DEBUG] [.rfxcom.internal.RFXComBinding:152  ] - Received command (item='Bell', state='14', class='class org.openhab.core.library.types.DecimalType')
14:39:17.590 [DEBUG] [.rfxcom.internal.RFXComBinding:181  ] - 1-executeCommand0
14:39:17.606 [DEBUG] [.rfxcom.internal.RFXComBinding:184  ] - 2-executeCommand0
14:39:17.613 [DEBUG] [.rfxcom.internal.RFXComBinding:186  ] - 3-executeCommand0
14:39:17.621 [DEBUG] [.rfxcom.internal.RFXComBinding:188  ] - 4-executeCommand0
14:39:17.627 [DEBUG] [.rfxcom.internal.RFXComBinding:190  ] - 5-executeCommand0

**Comment by AlexCPU** [comment]
Hi @Speederc 

Looks like progress, at least it's outputting more lines, but still not working.
```
2016-01-30 12:27:08.323 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command: DoorbellSounder 3
2016-01-30 12:27:08.324 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command (item='DoorbellSounder', state='3', class='class org.openhab.core.library.types.DecimalType')
2016-01-30 12:27:08.325 [DEBUG] [.rfxcom.internal.RFXComBinding] - 1-executeCommand0
2016-01-30 12:27:08.326 [DEBUG] [.rfxcom.internal.RFXComBinding] - 2-executeCommand0
2016-01-30 12:27:08.327 [DEBUG] [.rfxcom.internal.RFXComBinding] - 3-executeCommand0
2016-01-30 12:27:08.328 [DEBUG] [.rfxcom.internal.RFXComBinding] - 4-executeCommand0
2016-01-30 12:27:08.330 [DEBUG] [.rfxcom.internal.RFXComBinding] - 5-executeCommand0
```

LIGHTING5.LIGHTWAVERF works again though.  In case I hadn't made it clear, Lighting5 has been working in the first couple of builds you did, but stopped at some point:

```
2016-01-30 12:25:06.988 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command: LoungeLight OFF
2016-01-30 12:25:06.992 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command (item='LoungeLight', state='OFF', class='class org.openhab.core.library.types.OnOffType')
2016-01-30 12:25:06.993 [DEBUG] [.rfxcom.internal.RFXComBinding] - 1-executeCommand0
2016-01-30 12:25:06.994 [DEBUG] [.rfxcom.internal.RFXComBinding] - 2-executeCommand0
2016-01-30 12:25:06.996 [DEBUG] [.rfxcom.internal.RFXComBinding] - 3-executeCommand0
2016-01-30 12:25:06.997 [DEBUG] [.rfxcom.internal.RFXComBinding] - 4-executeCommand0
2016-01-30 12:25:06.998 [DEBUG] [.rfxcom.internal.RFXComBinding] - 5-executeCommand0
2016-01-30 12:25:06.999 [DEBUG] [.rfxcom.internal.RFXComBinding] - Transmitting data: 0A14000400000201000000
2016-01-30 12:25:07.000 [DEBUG] [.rfxcom.internal.RFXComBinding] - 6-executeCommand0
2016-01-30 12:25:08.192 [DEBUG] [rfxcom.internal.ResultRegistry] - Transmitter response received:
Raw data = 0402010400
 - Packet type = TRANSMITTER_MESSAGE
 - Seq number = 4
 - Sub type = UNKNOWN
 - Response = ACK
2016-01-30 12:25:08.200 [DEBUG] [.b.r.internal.RFXComConnection] - Data received:
Raw data = 0402010400
 - Packet type = TRANSMITTER_MESSAGE
 - Seq number = 4
 - Sub type = UNKNOWN
 - Response = ACK
2016-01-30 12:25:08.200 [DEBUG] [.rfxcom.internal.RFXComBinding] - Command succesfully transmitted, 'ACK' received
```

**Comment by Speederc** [comment]
Hi @AlexCPU & @McShannon 
I found a memory leak in the code, can you test this new version which should fix it :  http://dr-petelot-marion.chirurgiens-dentistes.fr/temp/org.openhab.binding.rfxcom_1.9.0.201601310054.jar ?
Thank you

**Comment by AlexCPU** [comment]
It works!  Thanks very much!

```
2016-01-31 18:01:44.587 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command: DoorbellSounder 3
2016-01-31 18:01:44.588 [DEBUG] [.rfxcom.internal.RFXComBinding] - Received command (item='DoorbellSounder', state='3', class='class org.openhab.core.library.types.DecimalType')
2016-01-31 18:01:44.589 [DEBUG] [.rfxcom.internal.RFXComBinding] - 1-executeCommand0
2016-01-31 18:01:44.589 [DEBUG] [.rfxcom.internal.RFXComBinding] - 2-executeCommand0
2016-01-31 18:01:44.590 [DEBUG] [.rfxcom.internal.RFXComBinding] - 3-executeCommand0
2016-01-31 18:01:44.590 [DEBUG] [.rfxcom.internal.RFXComBinding] - 4-executeCommand0
2016-01-31 18:01:44.591 [DEBUG] [.rfxcom.internal.RFXComBinding] - 5-executeCommand0
2016-01-31 18:01:44.592 [DEBUG] [.rfxcom.internal.RFXComBinding] - Transmitting data: 0716000300350300
2016-01-31 18:01:44.593 [DEBUG] [.rfxcom.internal.RFXComBinding] - 6-executeCommand0
2016-01-31 18:01:46.055 [DEBUG] [rfxcom.internal.ResultRegistry] - Transmitter response received:
Raw data = 0402010300
 - Packet type = TRANSMITTER_MESSAGE
 - Seq number = 3
 - Sub type = UNKNOWN
 - Response = ACK
2016-01-31 18:01:46.063 [DEBUG] [.rfxcom.internal.RFXComBinding] - Command succesfully transmitted, 'ACK' received
2016-01-31 18:01:46.063 [DEBUG] [.b.r.internal.RFXComConnection] - Data received:
Raw data = 0402010300
 - Packet type = TRANSMITTER_MESSAGE
 - Seq number = 3
 - Sub type = UNKNOWN
 - Response = ACK
```

**Comment by Speederc** [comment]
Great ! Thank you @AlexCPU and @McShannon for your testings and your perseverance !
All was due to a stupid bad array size (8 bytes instead of 9)...
It will be merged in a future PR.

**Comment by McShannon** [comment]
Thanks @Speederc, all working like a charm! Thanks for your efforts in getting this working. I am now off to annoy the household with random chimes.

**Comment by sheppy99** [comment]
Hi Guys,
Is this binding considered working for everything apart from "chime" now?
Is it is I'll take it for a drive and report back. I have various lighting 1 & 2 items both input and output plus an unused Byron SX.
Cheers
Kevin

Sent from my iPhone

On 1/02/2016, at 9:14 PM, McShannon <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks @Speederc<https://github.com/Speederc>, all working like a charm! Thanks for your efforts in getting this working. I am now off to annoy the household with random chimes.

-

**Comment by McShannon** [comment]
Hi @sheppy99 "chime" is now working for this binding. My ByronSX doorbell is receiving transmissions from OpenHab and playing whatever chime number i send it.

**Comment by sheppy99** [comment]
Hi McShannon,
Are the other types working too?
Cheers
Kevin

Sent from my iPhone

On 1/02/2016, at 9:22 PM, McShannon <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @sheppy99<https://github.com/sheppy99> "chime" is now working for this binding. My ByronSX doorbell is receiving transmissions from OpenHab and playing whatever chime number i send it.

-

**Comment by McShannon** [comment]
Hi @sheppy99 they are working well also.

**Comment by sheppy99** [comment]
Cool, I'll give it a try. Well done to all!
Currently busy troubleshooting Jessie images that have been stealing one of my GPIO Pins so decided to leave you guys to it :-)
Cheers
Kevin

Sent from my iPhone

On 1/02/2016, at 9:25 PM, McShannon <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @sheppy99<https://github.com/sheppy99> they are working well also.

-

**Comment by 9037568** [comment]
Fixed via merged PR #3954.

**Comment by McShannon** [comment]
Hey chaps, any of you got this working with OpenHAB2.
